Scripts/BuildPhases/LintAppLocalizedStringsUsage.swift

```diff
@@ -0,0 +1,324 @@
+protocol PBXObject: Decodable {}
+
+/// "Parent" type for PBXObjects referencing relative path information (`PBXFileReference`, `PBXGroup`)
+protocol PBXReference: PBXObject {
+    var name: String? { get }
+    var path: String? { get }
+    var sourceTree: Xcodeproj.SourceTree { get }
+}
+
+/// Types used to parse and decode the internals of an `.xcodeproj/project.pbxproj` file
+extension Xcodeproj {
+    /// An error `thrown` when an inconsistency is found while parsing the `.pbxproj` file.
+    enum DecodingError: Swift.Error, CustomStringConvertible {
+        case objectNotFound(id: ObjectUUID)
+        case unexpectedObjectType(id: ObjectUUID, expectedType: Any.Type, found: PBXObject)
+        var description: String {
+            switch self {
+                case .objectNotFound(id: let id):
+                    return "Unable to find object with UUID \(id)"
+                case .unexpectedObjectType(let id, let expectedType, let found):
+                    return  "Object with UUID \(id) was expected to be of type \(expectedType) but found \(found) instead."
+            }
+        }
+    }
+    
+    /// Type used to represent and decode the root object of a `.pbxproj` file.
+    struct PBXProjFile: Decodable {
+        let rootObject: ObjectUUID
+        let objects: [String: PBXObjectWrapper]
+
+        // Convenience methods
+
+        /// Returns the `PBXObject` instance with the given `ObjectUUID`, by looking it up in the list of `objects` registered in the project.
+        func object<T: PBXObject>(id: ObjectUUID) throws -> T {
+            guard let wrapped = objects[id] else { throw DecodingError.objectNotFound(id: id) }
+            guard let obj = wrapped.wrappedValue as? T else {
+                throw DecodingError.unexpectedObjectType(id: id, expectedType: T.self, found: wrapped.wrappedValue)
+            }
+            return obj
+        }
+
+        /// Returns the `PBXObject` instance with the given `ObjectUUID`, by looking it up in the list of `objects` registered in the project.
+        func object<T: PBXObject>(id: ObjectUUID) -> T? {
+            try? object(id: id) as T
+        }
+
+        /// The `PBXProject` corresponding to the `rootObject` of the project file.
+        var rootProject: PBXProject { try! object(id: rootObject) }
+
+        /// The `PBXGroup` corresponding to the main groop serving as root for the whole hierarchy of files and groups in the project.
+        var mainGroup: PBXGroup { try! object(id: rootProject.mainGroup) }
+
+        /// The list of `PBXNativeTarget` targets found in the project.
+        var nativeTargets: [PBXNativeTarget] { rootProject.targets.compactMap(object(id:)) }
+
+        /// The list of `PBXBuildFile` build file references included in a given target.
+        func buildFiles(for target: PBXNativeTarget) -> [PBXBuildFile] {
+            guard let sourceBuildPhase: PBXSourcesBuildPhase = target.buildPhases.lazy.compactMap(object(id:)).first else { return [] }
```

Comment:
The magic of Swift type system:
 - `sourceBuildPhase` being typed as `PBXSourceBuildPhase`…
 - … makes the type system infer that the whole `….compactMap(…).first` expression must be of type `PBXSourceBuildPhase?`…
 - … which means the result of `….compactMap(…)` must be of type `[PBXSourceBuildPhase]`…
 - … which implies that the function called inside the `compactMap` must return a `PBXSourceBuildPhase?` (and `compactMap` will only keep the returned values which are not `nil`)…
 - … which finally means that the generic type `T` for the `object(id:)` method specialization can be infered to be `T = PBXSourcesBuildPhase`…

This is also why, btw, this `compactMap` code filters to only go thru the `PBXSourcesBuildPhase` instances, ignoring / filtering out any potential other type of Build Phase that could be present in the `target.buildPhase` array 🙂 ✨